Neos.ContentRepository.NodeAccess/Classes/FlowQueryOperations/RemoveOperation.php

```diff
@@ -0,0 +1,78 @@
+<?php
+namespace Neos\ContentRepository\NodeAccess\FlowQueryOperations;
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\ContentRepository\Core\Projection\ContentGraph\Filter\FindReferencesFilter;
+use Neos\ContentRepository\Core\Projection\ContentGraph\Node;
+use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
+use Neos\Eel\FlowQuery\FlowQuery;
+use Neos\Eel\FlowQuery\OperationInterface;
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * Removes the given Node from the current context.
+ *
+ * The operation accepts one argument that may be an Array, a FlowQuery
+ * or an Object.
+ *
+ * !!! This is a Node specific implementation of the generic `remove` operation!!!
+ *
+ * The result is an array of {@see Node} instances.
+ *
+ * @api To be used in Fusion, in php this should be implemented directly
+ */
+final class RemoveOperation implements OperationInterface
+{
+    public function canEvaluate($context): bool
+    {
+        return count($context) === 0 || (isset($context[0]) && ($context[0] instanceof Node));
+    }
+
+    public function evaluate(FlowQuery $flowQuery, array $arguments): void
+    {
+        $nodeAggregateIdsToRemove = [];
+        if (isset($arguments[0])) {
+            if (is_iterable($arguments[0])) {
+                /** @var Node $node */
+                foreach ($arguments[0] as $node) {
+                    $nodeAggregateIdsToRemove[] = $node->nodeAggregateId->value;
+                }
+            } elseif ($arguments[0] instanceof Node) {
+                $nodeAggregateIdsToRemove[] = $arguments[0]->nodeAggregateId->value;
+            }
+        }
+
+        $filteredContext = array_filter(
+            $flowQuery->getContext(),
```

Comment:
`$context` seems to be of type `array|\Traversable` so `array_filter` might not always succeed (in case of `\Traversable`)

Reply:
Is adjusted